lib/appsignal/transaction.rb

```diff
@@ -133,22 +134,29 @@ def complete
 
       # If the transaction is a duplicate, we don't want to finish it,
       # because we want its finish time to be the finish time of the
-      # original transaction, and we do not want to set its sample
-      # data, because it should keep the sample data it duplicated from
-      # the original transaction.
-      # On duplicate transactions, the value of the sample flag, which
-      # is set on finish, will be duplicated from the original transaction.
-      sample_data if !is_duplicate && ext.finish(0)
-
-      # Ignore the first error as it is already set in the original
-      # transaction.
-      errors.drop(1).each do |error|
+      # original transaction.
+      # Duplicate transactions should always be sampled, as we only
+      # create duplicates for errors, which are always sampled.
+      should_sample = is_duplicate || ext.finish(0)
+
+      @error_blocks.each do |error, blocks|
+        # Ignore the error that is already set in this transaction.
+        next if error == @error_set
+
         duplicate.tap do |transaction|
-          transaction.set_error(error)
+          # In the duplicate transaction for each error, set an error
+          # with a block that calls all the blocks set for that error
+          # in the original transaction.
+          transaction.set_error(error) do
+            blocks.each { |block| block.call(transaction) }
+          end
+
           transaction.complete
         end
       end
 
+      @error_blocks[@error_set].each { |block| block.call(self) } if @error_set
+      sample_data if should_sample
```

Comment:
I do agree that it would not be a problem, but it would be redundant. We'd set sample data in the extension, duplicate it in the extension, then override that duplicate with a (maybe slightly modified) version of itself.